backend/dataall/aws/handlers/cloudformation.py

```diff
@@ -16,16 +16,16 @@ def __init__(self):
         pass
 
     @staticmethod
-    def client(AwsAccountId, region):
-        session = SessionHelper.remote_session(AwsAccountId)
+    def client(AwsAccountId, region, role=None):
+        session = SessionHelper.remote_session(accountid=AwsAccountId, role=role)
         return session.client('cloudformation', region_name=region)
 
     @staticmethod
-    def check_existing_cdk_toolkit_stack(AwsAccountId, region):
-        cfn = CloudFormation.client(AwsAccountId=AwsAccountId, region=region)
+    def check_existing_cdk_toolkit_stack(AwsAccountId, region, role=None):
         try:
+            cfn = CloudFormation.client(AwsAccountId=AwsAccountId, region=region, role=role)
             response = cfn.describe_stacks(StackName='CDKToolkit')
-        except cfn.exceptions.ClientError as e:
+        except ClientError as e:
             print(e)
             raise Exception('CDKToolkitNotFound')
```

Comment:
although you didn't touch this place in the code if I take a look at it closer there is a dead code:

` stacks = response['Stacks']
        if not len(stacks):
            raise Exception('CDKToolkitNotFound')`

This code will never be executed since if the stack with name 'CDKToolkit' doesn't exist describe_stacks always throws an error, it will not return an empty stacks list.

Have a look at the note in here: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/describe-stacks.html

'If the stack doesn’t exist, an ValidationError is returned.'

Reply:
I removed the dead lines of code :) Also, I think that in any case it will fail even before the describe-stacks call because the cfn client is trying to assume the cdk-look-up role

Reply:
indeed since the role will simply not be there